packages/sdk/src/evm/common/plugin.ts

```diff
@@ -189,11 +190,80 @@ export async function getMetadataForPlugins(
           ).map((fetchedMetadata) => fetchedMetadata.compilerMetadata);
         }
       } catch {}
+    } else {
+      throw new Error(
+        "No implementation found for Router. Can't fetch plugin metadata",
```

Comment:
are you sure we want to throw here? is not enough to return empty array?

Reply:
I added it temporarily, till we change the publish flow for dynamic contracts.
Right now we need at least one implementation, to be able to resolve the default extensions.

The empty array will eventually throw when encoding constructor params, but the messaging won't be clear.